### PR TITLE
UI support for hardware wallets

### PR DESCRIPTION
**Pull request by jacogr**
Closes https://github.com/ethcore/parity/issues/4266

https://www.youtube.com/watch?v=sWyQYZndiQY&feature=youtu.be

Initial support via `parity_hardwareAccountsInfo` & u2f Ledger interface. (Latter disabled by default, requires https support)

### CODE REVIEW

File: js/src/3rdparty/ledger/ledger.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        });
+        const rawTransaction = tx.serialize().toString('hex');
+
+        this._ledger.signTransaction(LEDGER_PATH_ETH, rawTransaction, (response, error) => {
+          if (error) {
+            reject(error);
+            return;
+          }
+
+          tx.v = new Buffer(response.v, 'hex');
```
Comment (derhuerst) [inline js/src/3rdparty/ledger/ledger.js]: Use `Buffer.from`, as `new Buffer` is deprecated.

File: js/src/3rdparty/ledger/ledger.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+      return new Promise((resolve, reject) => {
+        const chainId = new BigNumber(_chainId).toNumber();
+        const tx = new Transaction({
+          data: transaction.data || transaction.input,
+          gasPrice: numberToHex(transaction.gasPrice),
+          gasLimit: numberToHex(transaction.gasLimit),
+          nonce: numberToHex(transaction.nonce),
+          to: transaction.to ? transaction.to.toLowerCase() : undefined,
+          value: numberToHex(transaction.value),
+          v: new Buffer([chainId]), // pass the chainId to the ledger
```
Comment (derhuerst) [inline js/src/3rdparty/ledger/ledger.js]: see below.

**Comment by ngotchac** [comment]
(I just want to be able to actually test it with a hardware wallet ; so I guess next week when back in Berlin)

**Comment by jacogr** [comment]
As long as we make it for 1.6 - I have my Ledger all ready to go and prefer not keeping on swapping binaries to manage my own funds :) (As I have been doing)

**Comment by jacogr** [comment]
Lastest merged master - https://youtu.be/-ylCdV2Qv84

**Pull request merged**